src/esp/assets/ResourceManager.cpp

```diff
@@ -1192,13 +1193,16 @@ void ResourceManager::loadMaterials(Importer& importer,
     if (loadedAssetData.assetInfo.requiresLighting &&
         materialData->types() &
             Magnum::Trade::MaterialType::PbrMetallicRoughness) {
-      const Mn::Trade::PbrMetallicRoughnessMaterialData&
-          pbrMetallicRoughnessMaterialData =
-              static_cast<Mn::Trade::PbrMetallicRoughnessMaterialData&>(
-                  *materialData);
+      const auto& pbrMaterialData =
+          materialData->as<Mn::Trade::PbrMetallicRoughnessMaterialData>();
 
-      finalMaterial = gfx::buildPhongFromPbrMetallicRoughness(
-          pbrMetallicRoughnessMaterialData, textureBaseIndex, textures_);
+      if (flags_ & Flag::BuildPhongFromPbr) {
```

Comment:
BINGO: @bigbike we make guarantees that just be cause an object requires lighting that the scene actually has it: https://github.com/facebookresearch/habitat-sim/blob/77d201a15442e8bb6e68baca5d8552e81c763803/src/esp/assets/ResourceManager.cpp#L1070

Reply:
Aha, @bigbike figured it out. The Attribute Mangers are built BEFORE the lightSetups are actually changed in the getSimulator initialization so doing it here will always error until you call sim.setupLightKey after Simulator initialization!